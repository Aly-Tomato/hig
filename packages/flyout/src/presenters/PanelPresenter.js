import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import { ThemeContext } from "@hig/theme-context";
import { createCustomClassNames } from "@hig/utils";

import stylesheet from "./stylesheet";

export default function PanelPresenter({ children, onScroll, ...otherProps }) {
  const { className } = otherProps;
  const panelClassName = createCustomClassNames(className, "panel");

  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(
          { transitionStatus: null, anchorPoint: null },
          resolvedRoles
        );

        return (
          <div
            className={cx(css(styles.panel), panelClassName)}
            onScroll={onScroll}
          >
            {children}
          </div>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelPresenter.propTypes = {
  children: PropTypes.node,
  onScroll: PropTypes.func
};
