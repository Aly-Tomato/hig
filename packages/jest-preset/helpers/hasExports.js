module.exports = function hasExports(index, cases) {
  cases.forEach(({ name, value }) => {
    it(`exports ${name}`, () => {
      expect(index).toHaveProperty(name, value);
    });
  });
};
