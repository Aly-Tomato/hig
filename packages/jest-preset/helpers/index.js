/* eslint-disable global-require */
module.exports = {
  hasExports: require("./hasExports"),
  lastCallOfMock: require("./lastCallOfMock"),
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
