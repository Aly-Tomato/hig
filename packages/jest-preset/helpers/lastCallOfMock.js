module.exports = function lastCallOfMock(mockFn) {
  return mockFn.mock.calls[mockFn.mock.calls.length - 1];
};
