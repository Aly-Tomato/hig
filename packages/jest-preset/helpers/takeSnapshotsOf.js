const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const presenter = React.createElement(Component, props);
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
