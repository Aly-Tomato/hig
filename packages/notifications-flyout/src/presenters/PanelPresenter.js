import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import { Panel } from "@hig/flyout";
import ProgressRing from "@hig/progress-ring";
import ThemeContext from "@hig/theme-context";
import Typography from "@hig/typography";
import {
  UNMOUNTED,
  EXITED,
  ENTERING,
  ENTERED,
  EXITING
} from "react-transition-group/Transition";

import stylesheet from "./stylesheet";

export default function PanelPresenter({
  children,
  heading,
  innerRef,
  listMaxHeight,
  loadingTransitionState,
  onScroll,
  refListWrapper
}) {
  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet(resolvedRoles, {
          transitionState: null,
          loadingTransitionState
        });
        return (
          <Panel innerRef={innerRef}>
            <Typography elementType="header" style={styles.panelTitle}>
              {heading}
            </Typography>
            <section
              className={css(styles.panelContainer)}
              ref={refListWrapper}
              style={{ maxHeight: listMaxHeight }}
            >
              <div role="list" onScroll={onScroll}>
                {children}
              </div>
            </section>
            <footer className={css(styles.panelLoading)}>
              <ProgressRing size="s" />
            </footer>
          </Panel>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelPresenter.defaultProps = {
  heading: "Notifications"
};

PanelPresenter.propTypes = {
  children: PropTypes.node,
  heading: PropTypes.node,
  innerRef: PropTypes.func.isRequired,
  listMaxHeight: PropTypes.string,
  loadingTransitionState: PropTypes.oneOf([
    UNMOUNTED,
    EXITED,
    ENTERING,
    ENTERED,
    EXITING
  ]),
  onScroll: PropTypes.func,
  refListWrapper: PropTypes.func
};
