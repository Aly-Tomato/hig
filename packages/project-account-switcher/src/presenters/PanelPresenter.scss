.hig__project-account-switcher__panel {
  width: 200px;
  padding: 0px 10px 10px 10px;
}
