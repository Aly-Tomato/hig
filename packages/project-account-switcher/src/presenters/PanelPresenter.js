import React from "react";
import PropTypes from "prop-types";

import { Panel } from "@hig/flyout";

import "./PanelPresenter.scss";

export default function PanelPresenter({ innerRef, children }) {
  return (
    <Panel innerRef={innerRef}>
      <div className="hig__project-account-switcher__panel">{children}</div>
    </Panel>
  );
}

PanelPresenter.propTypes = {
  innerRef: PropTypes.func,
  children: PropTypes.node
};
