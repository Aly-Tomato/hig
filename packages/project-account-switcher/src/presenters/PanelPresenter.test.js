import { takeSnapshotsOf } from "@hig/jest-preset/helpers";
import PanelPresenter from "./PanelPresenter";

describe("project-account-switcher/presenters/PanelPresenter", () => {
  function handleRef() {}

  takeSnapshotsOf(PanelPresenter, [
    {
      description: "renders with minimal props",
      props: {
        innerRef: handleRef
      }
    },
    {
      description: "renders with all props",
      props: {
        children: "hello",
        innerRef: handleRef
      }
    }
  ]);
});
