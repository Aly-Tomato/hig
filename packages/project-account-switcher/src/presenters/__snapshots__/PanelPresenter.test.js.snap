// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project-account-switcher/presenters/PanelPresenter  1`] = `
<div
  className="hig__flyout-v1__panel-container"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="hig__flyout-v1__panel-container__inner"
  >
    <div
      className="hig__project-account-switcher__panel"
    />
  </div>
</div>
`;

exports[`project-account-switcher/presenters/PanelPresenter  2`] = `
<div
  className="hig__flyout-v1__panel-container"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="hig__flyout-v1__panel-container__inner"
  >
    <div
      className="hig__project-account-switcher__panel"
    >
      hello
    </div>
  </div>
</div>
`;
